commcare-cloud/commcare_cloud/commands/ansible/ansible_playbook.py

```diff
@@ -310,19 +318,133 @@ def make_parser(self):
+                for worker_name, details in celery_processes_list.items():
+                    # split comma separated names to individual workers
+                    for worker in worker_name.split(','):
+                        # ignore flower and celery periodic as celery workers
+                        if worker not in ['flower', 'celery_periodic']:
+                            if not celery_worker_config.get(worker):
+                                celery_worker_config[worker] = defaultdict(list)
+
+                            if details.get('num_workers', 1) > 2:
+                                for num in range(details.get('num_workers')):
+                                    full_worker_name = "%s-%s-celery_%s_%s" % (
+                                        project_name, args.environment, worker_name, num
+                                    )
+                                    celery_worker_config[worker][host].append(full_worker_name)
+                            else:
+                                full_worker_name = "%s-%s-celery_%s_%s" % (
+                                    project_name, args.environment, worker_name, 0
+                                )
+                                celery_worker_config[worker][host].append(full_worker_name)
+        return celery_worker_config
+
+    def get_workers_to_work_on(self, args):
+        """
+        :return:
+        workers_by_host: full name of workers per host to work on
+        """
+        celery_config = self.get_celery_config(args)
+        # full name of workers per host to run an action on
+        workers_by_host = defaultdict(list)
+        if args.worker_name:
+            for worker_name in args.worker_name.split(','):
+                if worker_name not in celery_config:
+                    puts(colored.red(
+                        "%s not found in the list of possible worker names, %s" % (
+                            worker_name, celery_config.keys()
+                        )))
+                    raise
+                for host, full_worker_names in celery_config[worker_name].items():
+                    workers_by_host[host].extend(full_worker_names)
+        else:
+            for worker_name in celery_config:
+                for host, full_worker_names in celery_config[worker_name].items():
+                    workers_by_host[host].extend(full_worker_names)
+        return workers_by_host
+
+    def run_for_celery(self, service_group, action, args, unknown_args):
+        exit_code = 0
+        service = "celery"
+        if action == "status" and not args.worker_name:
+            args.shell_command = "supervisorctl %s" % action
+            args.inventory_group = self.get_inventory_group_for_service(service, args.service_group)
+            exit_code = RunShellCommand(self.parser).run(args, unknown_args)
+        else:
+            try:
+                workers_by_host = self.get_workers_to_work_on(args)
+            except Exception:
```

Comment:
We try not to use `except Exception` where another more explicit option exists. What do you intend for this except to catch?

Reply:
in method `get_workers_to_work_on` in case the worker name passed as the argument in not found, we need this to return exit code 1. So one way i thought of doing this was to raise an exception in that method and then catch that here. I am going to make this not generic.
Another way of doing this is to check if that method returns no workers.. going to think more about it.

Reply:
this got removed in a later commit.